From: Laurent Bigonville <bigon@bigon.be>
Date: Fri, 25 Dec 2020 14:13:05 +0100
Subject: service: Stop the client for system applications when accuracy is
 set to 0

Now that the accuracy set by the agent is used again for system
applications, we should again stop the client when the accuracy is set
to 0
---
 src/gclue-service-client.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/gclue-service-client.c b/src/gclue-service-client.c
index 62ea932..3e68f55 100644
--- a/src/gclue-service-client.c
+++ b/src/gclue-service-client.c
@@ -330,7 +330,6 @@ on_agent_props_changed (GDBusProxy *agent_proxy,
         while (g_variant_iter_loop (iter, "{&sv}", &key, &value)) {
                 GClueAccuracyLevel max_accuracy;
                 const char *id;
-                gboolean system_app;
 
                 if (strcmp (key, "MaxAccuracyLevel") != 0)
                         continue;
@@ -338,8 +337,6 @@ on_agent_props_changed (GDBusProxy *agent_proxy,
                 gdbus_client = GCLUE_DBUS_CLIENT (client);
                 id = gclue_dbus_client_get_desktop_id (gdbus_client);
                 max_accuracy = g_variant_get_uint32 (value);
-                system_app = (gclue_client_info_get_xdg_id
-                              (client->priv->client_info) == NULL);
                 /* FIXME: We should be handling all values of max accuracy
                  *        level here, not just 0 and non-0.
                  */
@@ -354,8 +351,7 @@ on_agent_props_changed (GDBusProxy *agent_proxy,
                         start_client (client, accuracy);
                         g_debug ("Re-started '%s'.", id);
                 } else if (max_accuracy == 0 &&
-                           gclue_dbus_client_get_active (gdbus_client) &&
-                           !system_app) {
+                           gclue_dbus_client_get_active (gdbus_client)) {
                         stop_client (client);
                         client->priv->agent_stopped = TRUE;
                         g_debug ("Stopped '%s'.", id);
